fixed sneak pressed

leolee3914 opened the pull request:
<!-- Summarize your PR here. Keep it short and simple. -->
<!-- Explain existing problems or why this pull request is necessary -->

### Related issues & PRs
Related to #6867 

https://github.com/pmmp/PocketMine-MP/pull/6867/files#diff-f405c0ec75b0721cb66a250d6938b9267fd68285ffa443dfe8e0e156afac04e0R229

if both START_SNEAKING and STOP_SNEAKING are always false, any changes of SNEAKING is never handled and Player->isSneakPressed() is never changed
this happened when a player is holding shift to enter a 1.5 block space and then release shift

## Changes
### API changes
<!-- Any additions to the API that should be documented in release notes? -->
<!-- If not, you can delete this section -->
~~`Player->toggleSneak(bool $sneak, bool $sneakPressed = true) : bool` -> `Player->toggleSneak(?bool $sneak, bool $sneakPressed = true) : bool`~~

### Behavioural changes
<!-- Any change in how the server behaves, or its performance? -->
<!-- If not, you can delete this section -->

## Backwards compatibility
<!-- Any possible backwards incompatible changes? How are they solved, or how can they be solved? -->
<!-- If not, you can delete this section -->

## Follow-up
<!-- Suggest any actions to be done before/after merging this pull request -->
<!-- For example, future changes that this PR lays the groundwork for -->

## Tests
<!--
If this PR affects gameplay or user experience in some way, it must be manually tested.
Include any screenshots or videos of manual testing here.
Any test plugin code should also be pasted here if it can't be adapted to a PHPUnit test.
-->
i just tested the 4 states on #6548 with my own branch and haven't tested for PlayerToggleSneakEvent

leolee3914 committed:
fix

dktapps commented on src/player/Player.php:
```diff
@@ -2092,22 +2092,25 @@ public function toggleSprint(bool $sprint) : bool{
 
-	public function toggleSneak(bool $sneak, bool $sneakPressed = true) : bool{
-		if($sneak === $this->sneaking && $sneakPressed === $this->sneakPressed){
+	public function toggleSneak(?bool $sneak, bool $sneakPressed = true) : bool{
```
I don't think we should be changing this signature, especially not on the stable branch. You could do `?? $this->player->isSneaking()` in the packet handler instead, that should be sufficient and shouldn't require any changes to `Player`

dktapps reviewed (changes requested):
I see the issue, but I think the fix can be simplified

leolee3914 committed:
ok

leolee3914 committed:
Update Player.php

dktapps reviewed (dismissed):
Will need to test this myself to verify, but this looks good to me

dktapps committed:
remove unnecessary Player changes
this would only make a difference if someone un-cancelled PlayerToggleSneakEvent, and I don't see any reason why someone would do that if the sneak state didn't change.

dktapps commented:
Thanks!
